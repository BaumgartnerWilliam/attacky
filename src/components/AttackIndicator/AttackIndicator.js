import React from 'react';
import { Row, Spinner } from 'react-bootstrap';

const AttackIndicator = ({ isPlayer }) => (
  <Row data-testid="attacking" className="justify-content-center">
    <Spinner
      size="sm"
      animation="grow"
      variant={(isPlayer && 'primary') || 'danger'}
    />
  </Row>
);

export default AttackIndicator;
