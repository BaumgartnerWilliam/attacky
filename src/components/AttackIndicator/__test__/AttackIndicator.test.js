import React from 'react';
import { render } from '@testing-library/react';
import AttackIndicator from '../AttackIndicator';

describe('attack-indicator component', () => {
  it('renders without crashing', () => {
    const { container } = render(<AttackIndicator />);
    expect(container).toBeInTheDocument();
  });

  it('renders the primary variant', () => {
    const { getByTestId } = render(<AttackIndicator isPlayer={true} />);
    const el = getByTestId('attacking');
    expect(el.firstChild).toHaveClass('text-primary');
  });

  it('renders the danger variant', () => {
    const { getByTestId } = render(<AttackIndicator />);
    const el = getByTestId('attacking');
    expect(el.firstChild).toHaveClass('text-danger');
  });

  it('matches snapshot', () => {
    const { getByTestId } = render(<AttackIndicator />);
    const testid = getByTestId('attacking');
    expect(testid).toMatchSnapshot();
  });
});
