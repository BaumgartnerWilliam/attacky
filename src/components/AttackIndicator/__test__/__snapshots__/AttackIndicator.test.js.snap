// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attack-indicator component matches snapshot 1`] = `
<div
  class="justify-content-center row"
  data-testid="attacking"
>
  <div
    class="spinner-grow spinner-grow-sm text-danger"
  />
</div>
`;
